Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc

```diff
@@ -13,179 +13,89 @@
-#include "DetectorDescription/Core/interface/DDMaterial.h"
 
 #include "DataFormats/CTPPSAlignment/interface/RPAlignmentCorrectionData.h"
 
 using namespace std;
 
 //----------------------------------------------------------------------------------------------------
 
-DetGeomDesc::DetGeomDesc(nav_type navtype, GeometricEnumType type) : _ddd(std::move(navtype)), _type(type)
-{ 
-	DDCompactView cpv;
-	DDExpandedView ev(cpv);
-	ev.goTo(_ddd);
-	_params = ((ev.logicalPart()).solid()).parameters();
-	_trans = ev.translation();
-	_rot = ev.rotation();
-	_shape = ((ev.logicalPart()).solid()).shape();
-	_name = ((ev.logicalPart()).ddname()).name();
-	_parents = ev.geoHistory();
-	_volume   = ((ev.logicalPart()).solid()).volume();
-	_density  = ((ev.logicalPart()).material()).density();
-	_weight   = _density * ( _volume / 1000.); // volume mm3->cm3
-	_copy     = ev.copyno();
-	_material = ((ev.logicalPart()).material()).name().fullname();
-}
-
-//----------------------------------------------------------------------------------------------------
-
-DetGeomDesc::DetGeomDesc(DDExpandedView* fv, GeometricEnumType type) : _type(type)
-{
-	_ddd = fv->navPos();
-	_params = ((fv->logicalPart()).solid()).parameters();  
-	_trans = fv->translation();
-	_rot = fv->rotation();
-	_shape = ((fv->logicalPart()).solid()).shape();
-	_name = ((fv->logicalPart()).ddname()).name();
-	_parents = fv->geoHistory();
-	_volume   = ((fv->logicalPart()).solid()).volume();  
-	_density  = ((fv->logicalPart()).material()).density();
-	_weight   = _density * ( _volume / 1000.); // volume mm3->cm3
-	_copy     = fv->copyno();
-	_material = ((fv->logicalPart()).material()).name().fullname();
-}
-
-//----------------------------------------------------------------------------------------------------
-
-DetGeomDesc::DetGeomDesc(DDFilteredView* fv, GeometricEnumType type) : _type(type)
+DetGeomDesc::DetGeomDesc( DDFilteredView* fv )
 {
-	_ddd = fv->navPos();
-	_params = ((fv->logicalPart()).solid()).parameters();
-	_trans = fv->translation();
-	_rot = fv->rotation();
-	_shape = ((fv->logicalPart()).solid()).shape();
-	_name = ((fv->logicalPart()).ddname()).name();
-	_parents = fv->geoHistory();
-	_volume   = ((fv->logicalPart()).solid()).volume();
-	_density  = ((fv->logicalPart()).material()).density();
-	_weight   = _density * ( _volume / 1000.); // volume mm3->cm3
-	_copy     = fv->copyno();
-	_material = ((fv->logicalPart()).material()).name().fullname();
+  m_params = ((fv->logicalPart()).solid()).parameters();
+  m_trans = fv->translation();
+  m_rot = fv->rotation();
+  m_name = ((fv->logicalPart()).ddname()).name();
+  m_copy = fv->copyno();
+  m_z = fv->geoHistory().back().absTranslation().z();
 }
 
 //----------------------------------------------------------------------------------------------------
 
-DetGeomDesc::DetGeomDesc(const DetGeomDesc &ref)
+DetGeomDesc::DetGeomDesc( const DetGeomDesc &ref )
 {
-	(*this) = ref;
+  (*this) = ref;
 }
```

Comment:
Isn't this copy constructor the default one?

Reply:
done